### ISSUE TITLE
The experimental results were very poor.

### ISSUE THREAD

**Issue by luyui1478-blip** [niki-amini-naieni/CountGD#60]
eval：MAE: 7.508394851649129, RMSE: 31.545309696660947
test：MAE: 8.008907563025211, RMSE: 82.63245993880673

<img width="1532" height="899" alt="Image" src="https://github.com/user-attachments/assets/75f450f2-0374-47db-8b6e-902b91e3e801" />
I'm a complete beginner, and I followed the tutorial exactly, but I don't understand why I'm getting this result. Is there something wrong with my commands?

eval command：
```
nohup python -u main_inference.py \
    --output_dir ./gdino_val1 \
    -c config/cfg_fsc147_val.py \
    --eval \
    --datasets config/datasets_fsc147_val.json \
    --pretrain_model_path ./gdino_train1/checkpoint_best_regular.pth \
    --options text_encoder_type=checkpoints/bert-base-uncased \
    --crop \
    --sam_tt_norm \
    --remove_bad_exemplar \
    >> ./inference.log 2>&1 &
```
test command：
```
nohup python -u main_inference.py \
  --output_dir ./gdino_test1 \
  -c config/cfg_fsc147_test.py \
  --eval \
  --datasets config/datasets_fsc147_test.json \
  --pretrain_model_path gdino_train1/checkpoint_best_regular.pth \
  --options text_encoder_type=checkpoints/bert-base-uncased \
  --crop \
  --sam_tt_norm \
  --remove_bad_exemplar \
  >> test.log 2>&1 &
```

**Comment by niki-amini-naieni** [comment]
Hello, the issue may be with the cropping. We observe when retraining the model, the optimal crop threshold varies sometimes. Please see the issues below for further guidance and the information here: https://github.com/niki-amini-naieni/CountGD/blob/main/training.md:
https://github.com/niki-amini-naieni/CountGD/issues/32
https://github.com/niki-amini-naieni/CountGD/issues/16

For the published checkpoints, the intuitive crop threshold of 900 objects worked. However, for retrained checkpoints, the threshold seems to vary.

**Comment by niki-amini-naieni** [comment]
I think the main issue is that the Open-GroundingDINO code has some inherent non-determinism other than the seed, so the training code for CountGD (built on top of the Open-GroundingDINO codebase) is not perfectly reproducible. The RMSE in particular is extremely sensitive to outliers, and any data samples with higher than 900 objects for which adaptive cropping is not being applied may contribute significantly to this error. One way to check this would be to see if for all examples with <=900 objects are the MAE and RMSE the same for the published and retrained checkpoints. What if you apply the adaptive cropping to examples for which you know have more than 900 objects? Perhaps setting a softer adaptive cropping threshold would fix your issue.

**Comment by luyui1478-blip** [comment]
> I think the main issue is that the Open-GroundingDINO code has some inherent non-determinism other than the seed, so the training code for CountGD (built on top of the Open-GroundingDINO codebase) is not perfectly reproducible. The RMSE in particular is extremely sensitive to outliers, and any data samples with higher than 900 objects for which adaptive cropping is not being applied may contribute significantly to this error. One way to check this would be to see if for all examples with <=900 objects are the MAE and RMSE the same for the published and retrained checkpoints. What if you apply the adaptive cropping to examples for which you know have more than 900 objects? Perhaps setting a softer adaptive cropping threshold would fix your issue.

Thank you very much!I made the following changes, and the experimental results are consistent with those in the paper.

<img width="725" height="57" alt="Image" src="https://github.com/user-attachments/assets/446aa925-456b-4eca-8f56-5f62d0d02aea" />

**Comment by niki-amini-naieni** [comment]
Great! Happy to help!
